pyproject.toml

```diff
@@ -1,10 +1,259 @@
+    "service_identity >= 18.1.0",
+    "idna >= 2.4",
+]
+
+conch = [
+    "pyasn1",
+    "cryptography >= 2.6",
+    "appdirs >= 1.4.0",
+    "bcrypt >= 3.0.0",
+]
+
+conch-nacl = [
+    "twisted[conch]",
+    # Used to support Ed25519 keys on systems with OpenSSL < 1.1.1b
+    "PyNaCl",
+]
+conch_nacl = [
+    "twisted[conch-nacl]",
+]
+
+serial = [
+    "pyserial >= 3.0",
+    "pywin32 != 226; platform_system == 'Windows'",
+]
+
+http2 = [
+    "h2 >= 3.0, < 5.0",
+    "priority >= 1.1.0, < 2.0",
+]
+
+contextvars = [
+    "contextvars >= 2.4, < 3; python_version < '3.7'",
+]
+
+all-non-platform = [
+    "twisted[test,tls,conch,serial,http2,contextvars]",
+]
+all_non_platform = [
+    "twisted[all-non-platform]",
+]
+
+macos-platform = [
+    "twisted[all-non-platform]",
+    "pyobjc-core",
+    "pyobjc-framework-CFNetwork",
+    "pyobjc-framework-Cocoa",
+]
+macos_platform = [
+    "twisted[macos-platform]",
+]
+
+windows-platform = [
+    "twisted[all-non-platform]",
+    "pywin32 != 226",
+]
+windows_platform = [
+    "twisted[windows-platform]",
+]
+
+osx-platform = [
+    "twisted[macos-platform]",
+]
+osx_platform = [
+    "twisted[osx-platform]",
+]
+
+gtk-platform = [
+    "twisted[all-non-platform]",
+    "pygobject",
+]
+gtk_platform = [
+    "twisted[gtk-platform]",
+]
+
+mypy = [
+    "twisted[dev,all-non-platform,conch-nacl]",
+    "mypy==0.930",
+    "mypy-zope==0.3.4",
+    "types-setuptools",
+    "types-pyOpenSSL",
+]
+
+[project.scripts]
+cftp = "twisted.conch.scripts.cftp:run"
+ckeygen = "twisted.conch.scripts.ckeygen:run"
+conch = "twisted.conch.scripts.conch:run"
+mailmail = "twisted.mail.scripts.mailmail:run"
+pyhtmlizer = "twisted.scripts.htmlizer:run"
+tkconch = "twisted.conch.scripts.tkconch:run"
+trial = "twisted.scripts.trial:run"
+twist = "twisted.application.twist._twist:Twist.main"
+twistd = "twisted.scripts.twistd:run"
+
+[project.urls]
+Changelog = "https://github.com/twisted/twisted/blob/HEAD/NEWS.rst"
+Documentation = "https://docs.twistedmatrix.com/"
+Homepage = "https://twistedmatrix.com/"
+Issues = "https://twistedmatrix.com/trac/report"
+Source = "https://github.com/twisted/twisted"
+Twitter = "https://twitter.com/twistedmatrix"
+
+[tool.hatch.metadata]
+allow-ambiguous-features = true
+
+[tool.hatch.version]
+source = "code"
+path = "src/twisted/__init__.py"
+search-paths = ["src"]
+
+[tool.hatch.metadata.hooks.fancy-pypi-readme]
+content-type = "text/x-rst"
+fragments = [
+    { path = "README.rst" },
+]
+
+[[tool.hatch.metadata.hooks.fancy-pypi-readme.substitutions]]
+# Munge links of the form `NEWS <NEWS.rst>`_ to point at the appropriate
+# location on GitHub so that they function when the long description is
+# displayed on PyPI.
+pattern = '`([^`]+)\s+<(?!https?://)([^>]+)>`_'
+replacement = '`\1 <https://github.com/twisted/twisted/blob/trunk/\2>`_'
+ignore-case = true
+
+[tool.hatch.build.targets.wheel]
+packages = [
+    "src/twisted",
+]
+exclude = [
+    "*.c",
```

Comment:
is this needed ?
Do we have .C or .H files?

Reply:
Dunno, I just copied `MANIFEST.in`. Removed